osf_tests/factories.py

```diff
@@ -505,43 +507,68 @@ class Meta:
         model = models.PreprintProvider
 
 
+def sync_set_identifiers(preprint):
+    ezid_return_value ={
+        'response': {
+            'success': '{doi}osf.io/{guid} | {ark}osf.io/{guid}'.format(
+                doi=settings.DOI_NAMESPACE, ark=settings.ARK_NAMESPACE, guid=preprint._id
+            )
+        },
+        'already_exists': False
+    }
+    preprint.set_preprint_identifiers(ezid_return_value)
```

Comment:
If you want to save some typing, then change `set_preprint_identifers` to take values.

```
preprint.set_identifier_values(doi=..., ark=...)
```

That way, you could do:

```python
identifiers = parse_identifiers(ezid_response)
preprint.set_identifier_values(**identifiers)
```